src/Mod/Measure/Gui/DlgPrefsMeasureAppearanceImp.cpp

```diff
@@ -0,0 +1,78 @@
+ /**************************************************************************
+ *   Copyright (c) 2023 Wanderer Fan <wandererfan@gmail.com>               *
+ *                                                                         *
+ *   This file is part of the FreeCAD CAx development system.              *
+ *                                                                         *
+ *   This library is free software; you can redistribute it and/or         *
+ *   modify it under the terms of the GNU Library General Public           *
+ *   License as published by the Free Software Foundation; either          *
+ *   version 2 of the License, or (at your option) any later version.      *
+ *                                                                         *
+ *   This library  is distributed in the hope that it will be useful,      *
+ *   but WITHOUT ANY WARRANTY; without even the implied warranty of        *
+ *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the         *
+ *   GNU Library General Public License for more details.                  *
+ *                                                                         *
+ *   You should have received a copy of the GNU Library General Public     *
+ *   License along with this library; see the file COPYING.LIB. If not,    *
+ *   write to the Free Software Foundation, Inc., 59 Temple Place,         *
+ *   Suite 330, Boston, MA  02111-1307, USA                                *
+ *                                                                         *
+ ***************************************************************************/
+
+#include "PreCompiled.h"
+
+#include "DlgPrefsMeasureAppearanceImp.h"
+#include "ui_DlgPrefsMeasureAppearanceImp.h"
+
+using namespace MeasureGui;
+
+DlgPrefsMeasureAppearanceImp::DlgPrefsMeasureAppearanceImp( QWidget* parent )
+  : PreferencePage( parent )
+  , ui(new Ui_DlgPrefsMeasureAppearanceImp)
+{
+    ui->setupUi(this);
+}
+
+DlgPrefsMeasureAppearanceImp::~DlgPrefsMeasureAppearanceImp()
+{
+    // no need to delete child widgets, Qt does it all for us
+}
+
+void DlgPrefsMeasureAppearanceImp::saveSettings()
+{
+    ui->cbMirror->onSave();
+    ui->dsbDistFactor->onSave();
+    ui->sbFontSize->onSave();
+    ui->cbText->onSave();
+    ui->cbLine->onSave();
+    ui->cbBackground->onSave();
+}
+
+void DlgPrefsMeasureAppearanceImp::loadSettings()
+{
+    ui->sbFontSize->onRestore();
+    ui->cbText->onRestore();
+    ui->cbBackground->onRestore();
+    ui->cbLine->onRestore();
+    ui->cbMirror->onRestore();
+    ui->dsbDistFactor->onRestore();
+}
+
+/**
+ * Sets the strings of the subwidgets using the current language.
+ */
+void DlgPrefsMeasureAppearanceImp::changeEvent(QEvent *e)
+{
+    if (e->type() == QEvent::LanguageChange) {
+        saveSettings();
```

Comment:
Why does it have to save and restore settings when changing the language?

Reply:
cut & paste code.  save/restore removed.  will look for original source, that probably needs changing as well.